mathesar_ui/src/components/SchemaName.svelte

```diff
@@ -4,6 +4,34 @@
   import NameWithIcon from './NameWithIcon.svelte';
 
   export let schema: SchemaEntry;
+  export let isLocked = false;
 </script>
 
-<NameWithIcon icon={iconSchema}>{schema.name}</NameWithIcon>
+<div class="container" class:is-locked={isLocked}>
+  <NameWithIcon icon={iconSchema}>{schema.name}</NameWithIcon>
+</div>
+
+<style lang="scss">
+  // TODO: Is there a better way to do style
+  // the icon and name inside NameWithIcon component?
+  :global(.container .icon) {
+    width: 1.71rem;
+    height: 1.71rem;
+    border-radius: var(--border-radius-m);
+    background-color: var(--brand-500);
+    color: var(--white);
+    padding: 0.3rem;
+    opacity: 1;
+  }
+
+  :global(.container.is-locked .icon) {
+    background-color: var(--slate-800);
+  }
+
+  :global(.container .name) {
+    font-size: 1.428rem;
+    font-weight: normal;
+    color: var(--slate-800);
+    margin-left: 0.25rem;
+  }
```

Comment:
`global` tag should not be on the outermost scope in a component. This will make all the styles here global eg., `.container .icon` will be a globally available css.

Instead change this to:
```
.container {
  :global(.icon) { ... }
}
```

Reply:
I'd like to take this critique a step further...

First of all, I agree with @pavish. Yes, `.container :global(.icon)` is better than `:global(.container .icon)`.

But also, `.container :global(.icon)` is still using global CSS which we ought to be using only as a last resort. Global CSS breaks encapsulation which I think hinders maintainability.

I added 45ec13325aa77328b2a05e5389de23626e82e322 to demonstrate the approach I would take here.

1. I made the low level component, `NameWithIcon`, slightly more flexible. Now it accepts a prop to configure whether or not it renders its icon within a rounded box. It also accepts a new CSS variable, `--icon-opacity`.

1. The mid-level component, `SchemaName`, now sets the color of the icon based on whether the schema is locked. It sets the icon opacity but does _not_ set the font size because it doesn't know where it will be rendered.

1. The high-level component, `SchemaRow`, specifies the font size and the boxed icon.

This way all the concerns are separated to their various levels so that the lower level components can be reused and refactored without worrying about what happens outside them.